File: 01_arrays_hashing/01_duplicate_integer/solution.py
"""
Arrays & Hashing -> Duplicate Integer
Neetcode 1 / 150
Leetcode 217

Python

Comments:
Me love python.

Time: O(n) loop through list of nums
Space: O(n) create a set of max size equal to list size
"""


class Solution:
    def hasDuplicate(self, nums: list[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

File: 01_arrays_hashing/01_duplicate_integer/solution.js
/*
Arrays & Hashing -> Duplicate Integer
Neetcode 1 / 150
Leetcode 217

Javascript

Comments:
Didn't realize Javascript has a Set object. That's nice, I guess.

Time: O(n) loop through list of nums
Space: O(n) create a set of max size equal to list size
*/

class Solution {
	/**
	 * @param {number[]} nums
	 * @return {boolean}
	 */
	hasDuplicate(nums) {
		const seen = new Set();
		for (const num of nums) {
			if (seen.has(num)) return true;
			seen.add(num);
		}
		return false;
	}
}
